src/NumoBundle/Entity/Published.php

```diff
@@ -43,13 +41,40 @@ class Published
     private $author;
 
     /**
+     * @var \DateTime
+     *
+     * @ORM\Column(name="authorUpdateDate", type="datetime")
+     */
+    private $authorUpdateDate;
+
+    /**
      * @ORM\ManyToOne(targetEntity="User", inversedBy="moderations")
      *
      */
     private $moderator;
 
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(name="moderatorUpdateDate", type="datetime")
+     */
+    private $moderatorUpdateDate;
 
 
+    public function __construct(Event $event, string $uid, $moderator = null)
+    {
+        $this
+            ->setDeleted(0)
+            ->setUid($uid)
+            ->setAuthor($event->getAuthor())
+            ->setAuthorUpdateDate($event->getCreationDate());
+        if (isset($moderator)) {
```

Comment:
as tu besoin de passer le modérateur en param ? n'est ce pas toujours la personne actuellement connecté donc $this->getUser ? et traiter donc un $publication->setModerateur($this->getUser())